Client/Frontend/Reader/ReaderMode.swift

```diff
@@ -29,6 +29,25 @@ enum ReaderModeTheme: String {
     case light = "light"
     case dark = "dark"
     case sepia = "sepia"
+
+    init() {
+        self = ReaderModeTheme(rawValue: ThemeManager.instance.currentName.rawValue)?.preferredTheme() ?? .light
+    }
+}
+
+extension ReaderModeTheme {
+    func preferredTheme(for theme: ReaderModeTheme? = nil) -> ReaderModeTheme {
+        let theme = theme ?? self
+        let appWideTheme = ReaderModeTheme(rawValue: ThemeManager.instance.currentName.rawValue) ?? .light
+        
+        if appWideTheme == .dark {
+            return .dark // app-wide dark overrides all
+        } else if theme != .sepia {
+            return .light
+        }
+
+        return .sepia
+    }
```

Comment:
Hi @nick130586, thanks for submitting the PR for the issue. I think the overall approach is good but we can make it even better by adding few comments and cleanup. We don't really need an extension and an init(). We can easily clean this up by using static function. What do you think? 

Eg. 
```Swift

enum ReaderModeTheme: String {
    case light = "light"
    case dark = "dark"
    case sepia = "sepia"
    
    static func preferredTheme(for theme: ReaderModeTheme? = nil) -> ReaderModeTheme {
        // If there is no reader theme provided than we default to light theme
        let readerTheme = theme ?? .light
        // Get current Firefox theme (Dark vs Normal)
        // Normal means light theme. This is the overall theme used
        // by Firefox iOS app
        let appWideTheme = ThemeManager.instance.currentName
        // We check for 3 basic themes we have Light / Dark / Sepia
        // Theme: Dark - app-wide dark overrides all
        if appWideTheme == .dark {
            return .dark
        // Theme: Sepia - special case for when the theme is sepia.
        // For this we only check the them supplied and not the app wide theme
        } else if readerTheme == .sepia {
            return .sepia
        }
        // Theme: Light - Default case for when there is no theme supplied i.e. nil and we revert to light
        return readerTheme
    }
} 

```

Reply:
```Swift 
//After refactoring we'll also change reader mode theme to use  `ReaderModeTheme.preferredTheme()`

readerModeStyle.theme = ReaderModeTheme.preferredTheme()
```